service/src/test/java/cz/muni/fi/pa165/service/ManufacturerServiceTest.java

```diff
@@ -0,0 +1,207 @@
+
+    @Test
+    public void findAll() {
+        List<Manufacturer> list = Arrays.asList(manufacturer1, manufacturer2);
+        when(manufacturerDAO.findAll()).thenReturn(list);
+        List<Manufacturer> result = manufacturerService.findAll();
+
+        Assert.assertNotNull(result);
+        Assert.assertEquals(list, result);
+    }
+
+    @Test
+    public void findAllEmpty() {
+        when(manufacturerDAO.findAll()).thenReturn(Collections.emptyList());
+        List<Manufacturer> result = manufacturerService.findAll();
+
+        Assert.assertNotNull(result);
+        Assert.assertEquals(result.size(), 0);
+    }
+
+    @Test
+    public void findByName() {
+        when(manufacturerDAO.findByName(manufacturer1.getName())).thenReturn(manufacturer1);
+        Manufacturer result = manufacturerService.findByName(manufacturer1.getName());
+
+        Assert.assertNotNull(result);
+        Assert.assertEquals(result, manufacturer1);
+    }
+
+    @Test
+    public void findByNameNonExisting() {
+        final String nonExistingName = "Manufacturer NonExisting";
+        when(manufacturerDAO.findByName(nonExistingName)).thenReturn(null);
+        Manufacturer result = manufacturerService.findByName(nonExistingName);
+
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void hasToxicProductionByIdNoBottles() {
+        when(manufacturerDAO.findOne(manufacturer1.getId())).thenReturn(manufacturer1);
+        when(bottleService.getAllBottlesFromManufacturer(manufacturer1)).thenReturn(Collections.emptyList());
+        boolean result = manufacturerService.hasToxicProduction(manufacturer1.getId());
+
+        Assert.assertNotNull(result);
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    public void hasToxicProductionByIdFalse() {
+        List<Bottle> bottleList = Arrays.asList(bottle1, bottle2);
+        when(manufacturerDAO.findOne(manufacturer1.getId())).thenReturn(manufacturer1);
+        when(bottleService.getAllBottlesFromManufacturer(manufacturer1)).thenReturn(bottleList);
+        boolean result = manufacturerService.hasToxicProduction(manufacturer1.getId());
+
+        Assert.assertNotNull(result);
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    public void hasToxicProductionByIdTrue() {
+        bottle1.setToxic(true);
+        List<Bottle> bottleList = Arrays.asList(bottle1, bottle2);
+        when(manufacturerDAO.findOne(manufacturer1.getId())).thenReturn(manufacturer1);
+        when(bottleService.getAllBottlesFromManufacturer(manufacturer1)).thenReturn(bottleList);
+        boolean result = manufacturerService.hasToxicProduction(manufacturer1.getId());
+
+        Assert.assertNotNull(result);
+        Assert.assertTrue(result);
+    }
+
+    @Test(expected = ServiceException.class)
+    public void hasToxicProductionByIdNonExisting() {
+        when(manufacturerDAO.findOne(manufacturer1.getId())).thenReturn(null);
+        manufacturerService.hasToxicProduction(manufacturer1.getId());
+    }
```

Comment:
Tady tento test neprojde. Service vraci false, ale nevim, jestli by to nemelo radsi vyhodit nejakou vyjimku?